router/io-spec.gobra

```diff
@@ -99,18 +99,18 @@ pure func asidsBefore(
 	prevSegLen int,
 	consDir bool,
 	asid io.IO_as) (res option[seq[io.IO_as]]) {
-	return let next_asid := (unfolding acc(sl.AbsSlice_Bytes(raw, 0, len(raw)), _) in asidFromIfs(dp, raw, numINF, currHFIdx, !consDir, asid)) in
+	return currHFIdx == prevSegLen ? some(seq[io.IO_as]{asid}) :
+		let next_asid := (unfolding acc(sl.AbsSlice_Bytes(raw, 0, len(raw)), _) in asidFromIfs(dp, raw, numINF, currHFIdx, !consDir, asid)) in
```

Comment:
maybe we could have `acc(sl.AbsSlice_Bytes(raw, 0, len(raw))` in the precondition of `asidFromIfs`, I think in general it would look better than having an unfold in every call

Reply:
also, in Go, we should use call case: `next_asid` should be `nextAsId` or `nextAsid`

Reply:
> maybe we could have `acc(sl.AbsSlice_Bytes(raw, 0, len(raw))` in the precondition of `asidFromIfs`, I think in general it would look better than having an unfold in every call

This way has the advantage that the precondition of `asidFromIfs` automatically triggers the precondition of  `binary.BigEndian.Uint16(raw[idx+4:idx+6])`. Otherwise I would need something like `forall i :: raw[idx+4:idx+6] == raw[idx+4+i]`.



> also, in Go, we should use call case: `next_asid` should be `nextAsId` or `nextAsid`

Yes, I will change that. Sorry, I always switch up the style.